src/python/pants/engine/target.py

```diff
@@ -578,6 +563,46 @@ def types(self) -> Tuple[Type[Target], ...]:
         return tuple(self.aliases_to_types.values())
 
 
+@dataclass(frozen=True)
+class PluginField:
```

Comment:
> When backends register Targets, we don't call them PluginTargets, etc.

Yes, but when adding a new target, we aren't modifying anything pre-existing. 

The only reason for it to be called `PluginField` is because it is adding a new field to something pre-existing. By definition, it couldn't have been included in the original target definition's class property.

Once it's registered, all of the Target methods treat it as first-class.